src/optimized.tsx

```diff
@@ -1,28 +1,30 @@
 import * as React from 'react';
 
-import { getComponent, getComponentFromCache, isCached } from './map';
+import { cache, getComponent } from './map';
 import { mapLoadable } from './bundler';
 
 type Props<T> = T & {
   screenName: string;
 };
 
 const Suspender = React.forwardRef(function <T extends {}>({ screenName, ...rest }: Props<T>, ref: React.Ref<any>) {
-  const Component = isCached(screenName) ? getComponentFromCache(screenName).component : null;
+  const isCached = cache.has(screenName);
 
-  if(!Component) {
+  if(!isCached) {
     throw getComponent(screenName);
   }
 
+ const { component: Component } = cache.get(screenName)!;
+
   return <Component ref={ref} {...rest} />;
 })
 
 const optimized = <T extends object>(screenName: string): React.ForwardRefExoticComponent<React.PropsWithoutRef<T> & React.RefAttributes<unknown>> => {
   return React.forwardRef((props: T, ref) => {
-    const { placeholder } = mapLoadable[screenName];
+    const { placeholder } = mapLoadable.get(screenName) || {};
```

Comment:
Is there really a chance that we receive nothing from the cache?..

Reply:
no, but typescript says yes